test/spec/util/util.js

```diff
@@ -225,4 +225,32 @@ describe('iD.util', function() {
             expect(iD.utilUnicodeCharsTruncated('😎😬😆😵😴😄🙂🤔', 255)).to.eql('😎😬😆😵😴😄🙂🤔');
         });
     });
+
+    describe('utilNormalizeDateString', function() {
+        it('pads dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-01-1').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('1970-1-01').value).to.eql('1970-01-01');
+            expect(iD.utilNormalizeDateString('-1-1-01').value).to.eql('-0001-01-01');
+            expect(iD.utilNormalizeDateString('123').value).to.eql('0123');
+            expect(iD.utilNormalizeDateString('-4003').value).to.eql('-4003'); // beyond displayable year range but still valid
+            expect(iD.utilNormalizeDateString('31337').value).to.eql('31337');
+            expect(iD.utilNormalizeDateString('-31337').value).to.eql('-31337');
+        });
+        it('wraps dates', function() {
+            expect(iD.utilNormalizeDateString('1970-01-00').value).to.eql('1969-12-31');
+            expect(iD.utilNormalizeDateString('1970-12-32').value).to.eql('1971-01-01');
+            expect(iD.utilNormalizeDateString('1970-02-29').value).to.eql('1970-03-01');
+            expect(iD.utilNormalizeDateString('1970-00').value).to.eql('1969-12');
+            expect(iD.utilNormalizeDateString('1970-23').value).to.eql('1971-11'); // no EDTF for now
```

Comment:
I’ve already make the mistake of entering EDTF year-season combinations into `start_date` and `end_date` a few times. It’ll be nice to get warned about this mistake instead of having to notice that the time slider never filters the feature out.